### PR TITLE
add documentation describing the difference in api response status code.

### PR DESCRIPTION
**Pull request by allencloud**
this pr adds a documentation describing the difference between Swarm and Docker Engine in the aspect of  api response status code.
Four methods are included, and they are GET, POST, PUT and DELETE.
The comparison is based on api v1.22, and all Docker Status Codes in api v1.22 are referenced from [docker-remote-api-v1.22]

Signed-off-by: Sun Hongliang <allen.sun@daocloud.io>

### CODE REVIEW

**Comment by nishanttotla** [comment]
Thanks @allencloud. We'll need some time to go through this.

ping @moxiegirl

**Comment by allencloud** [comment]
@nishanttotla @moxiegirl 
I think it is not easy to go through this, and there must be some case I did not make clear.

Thanks.

**Comment by jimmyxian** [comment]
@allencloud Cool, nice work. Thanks

**Comment by moxiegirl** [comment]
@allencloud Thank you for the contribution.  Our documentation build is broken because your files is missing some critical components.  Please go ahead and add them,

File: docs/status-code-comparison-to-docker.md [hunk 1]
```diff
@@ -0,0 +1,595 @@
+This is a documentation describing the difference between Swarm and Docker Engine in the aspect of  api response status code.
```
Comment (moxiegirl) [inline docs/status-code-comparison-to-docker.md]: Add metadata

```
<!--[metadata]>
+++
title = "API response codes"
description = "Swarm API response codes"
keywords = ["docker, swarm, response, code,  api"]
[menu.main]
parent="workw_swarm"
weight=99
+++
<![end-metadata]-->
```

Comment (moxiegirl) [inline docs/status-code-comparison-to-docker.md]: Please add an H1:

```
# Understand the Swarm vs. Engine response codes
```

File: docs/status-code-comparison-to-docker.md [hunk 2]
```diff
@@ -0,0 +1,595 @@
...
+
+Five methods are included, and they are HEAD, GET, POST, PUT and DELETE.
+
+The comparison is based on api v1.22, and all Docker Status Codes in api v1.22 are referenced from [docker-remote-api-v1.22](https://github.com/docker/docker/blob/master/docs/reference/api/docker_remote_api_v1.22.md
+).
+
+#HEAD
```
Comment (moxiegirl) [inline docs/status-code-comparison-to-docker.md]:
```
## HEAD
```

**Comment by allencloud** [comment]
@moxiegirl 
I have already added the missing part. Thanks for your remind. :)

File: docs/status-code-comparison-to-docker.md [hunk 3]
```diff
@@ -0,0 +1,597 @@
...
+keywords = ["docker, swarm, response, code,  api"]
+[menu.main]
+parent="workw_swarm"
+weight=99
++++
+<![end-metadata]-->
+
+This is a documentation describing the difference between Swarm and Docker Engine in the aspect of  api response status codes.
+
+# Understand the Swarm vs. Engine response codes
```
Comment (moxiegirl) [inline docs/status-code-comparison-to-docker.md]: This should be at line 11 --- start with the head

File: docs/status-code-comparison-to-docker.md [hunk 4]
```diff
@@ -0,0 +1,597 @@
...
+
+Four methods are included, and they are GET, POST, PUT and DELETE.
+
+The comparison is based on api v1.22, and all Docker Status Codes in api v1.22 are referenced from [docker-remote-api-v1.22](https://github.com/docker/docker/blob/master/docs/reference/api/docker_remote_api_v1.22.md
+).
+
+#GET
```
Comment (moxiegirl) [inline docs/status-code-comparison-to-docker.md]: These should be second level heads and need a space between the `##` and the first letter.

#GET > ## GET
#POST > ## Post

**Comment by allencloud** [comment]
I found that docker-remote-api-v1.22.md is not correct at any api response status code.
Since in docker engine's source code, not all api returns a status code of 500.

**Comment by allencloud** [comment]
WDYT @moxiegirl

File: docs/status-code-comparison-to-docker.md [hunk 5]
```diff
@@ -0,0 +1,597 @@
...
+description = "Swarm API response codes"
+keywords = ["docker, swarm, response, code,  api"]
+[menu.main]
+parent="workw_swarm"
+weight=99
++++
+<![end-metadata]-->
+# Understand the Swarm vs. Engine response codes
+
+This is a documentation describing the difference between Swarm and Docker Engine in the aspect of  api response status codes.
```
Comment (moxiegirl) [inline docs/status-code-comparison-to-docker.md]: @allencloud Thanks for the contribution.  I think the description should include a bit more detail.  Not everyone coming to the page will know that they share the same API.  A bit of background is helpful.

https://gist.github.com/moxiegirl/33030590a9a32fe0c0c7

Docker Engine provides a REST API for making calls to the Engine daemon. Docker
Swarm allows a caller to make the same calls to a cluster of Engine daemons.
While the API calls are the same, the API response status codes do differ. This
document explains the differences.

Four methods are included, and they are GET, POST, PUT and DELETE.

The comparison is based on api v1.22, and all Docker Status Codes in api v1.22
are referenced from the [Engine remote API
v1.22](https://docs.docker.com/engine/reference/api/docker_remote_api_v1.22/).

**Comment by allencloud** [comment]
Thanks @moxiegirl  That is reasonable.

File: docs/status-code-comparison-to-docker.md (hunk 5 above)
Comment (moxiegirl) [inline docs/status-code-comparison-to-docker.md]: @allencloud Can you remove line 13? It isn't needed any more.

**Comment by allencloud** [comment]
@moxiegirl 
I changed the line you methioned.

**Comment by moxiegirl** [comment]
Thank you @allencloud LGTM!  @nishanttotla for a final review.

**Comment by allencloud** [comment]
Any update? @nishanttotla

File: docs/status-code-comparison-to-docker.md [hunk 6]
```diff
@@ -0,0 +1,597 @@
...
+
+The comparison is based on api v1.22, and all Docker Status Codes in api v1.22 are referenced from [docker-remote-api-v1.22](https://github.com/docker/docker/blob/master/docs/reference/api/docker_remote_api_v1.22.md
+).
+
+## GET
+
+- Route:          /_ping
+- Handler:        ping
+
+|Swarm Status Code:|Docker Status Code|
```
Comment (nishanttotla) [inline docs/status-code-comparison-to-docker.md]: Can you remove the colon (:) next to each `Swarm Status Code`?

File: docs/status-code-comparison-to-docker.md [hunk 7]
```diff
@@ -0,0 +1,597 @@
...
+Docker Engine provides a REST API for making calls to the Engine daemon. Docker Swarm allows a caller to make the same calls to a cluster of Engine daemons. While the API calls are the same, the API response status codes do differ. This document explains the differences.
+
+Four methods are included, and they are GET, POST, PUT and DELETE.
+
+The comparison is based on api v1.22, and all Docker Status Codes in api v1.22 are referenced from [docker-remote-api-v1.22](https://github.com/docker/docker/blob/master/docs/reference/api/docker_remote_api_v1.22.md
+).
+
+## GET
+
+- Route:          /_ping
```
Comment (nishanttotla) [inline docs/status-code-comparison-to-docker.md]: /_ping -> `/_ping`

File: docs/status-code-comparison-to-docker.md [hunk 8]
```diff
@@ -0,0 +1,597 @@
...
+- Handler:        ping
```
Comment (nishanttotla) [inline docs/status-code-comparison-to-docker.md]: ping -> `ping`

**Comment by nishanttotla** [comment]
@allencloud this is great. Just some minor things, after that we should be able to merge it.

File: docs/status-code-comparison-to-docker.md (hunk 8 above)
Comment (allencloud) [inline docs/status-code-comparison-to-docker.md]: Thanks @nishanttotla 
add `` to every Route and every Handler

File: docs/status-code-comparison-to-docker.md (hunk 6 above)
Comment (allencloud) [inline docs/status-code-comparison-to-docker.md]: thanks for your careful code review. changed that

**Comment by nishanttotla** [comment]
LGTM

**Comment by nishanttotla** [comment]
Thanks @allencloud!

**Comment by nishanttotla** [comment]
@allencloud test seems to fail.
@moxiegirl is this relevant?

**Comment by allencloud** [comment]
Yes, I am wondering what can cause this.

**Pull request merged**